client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx

```diff
@@ -125,6 +219,14 @@ const SiteSetupList = ( {
 		}
 	}, [ currentTaskId, dispatch, tasks ] );
 
+	// If specified then automatically complete the current task when viewed.
+	useEffect( () => {
+		if ( currentTask?.completeOnView ) {
```

Comment:
It seems that we can use `if ( currentTask?.completeOnView && ! currentTask.isCompleted ) {` instead of `if ( currentTask?.completeOnView ) {` so that we avoid running again the same code (and especially api calls) even for completed tasks. Tested locally and works, would like your opinion though.

Reply:
Fixed.